src/app/Services/shop-list.service.ts

```diff
@@ -1,23 +1,25 @@
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ShopListService {
-  ingredients = [
+  ingredients = new BehaviorSubject<Array<any>>([
     { id: 1, name: 'Bread', amount: 4 },
     { id: 2, name: 'Tomato', amount: 6 },
     { id: 3, name: 'Vegetable', amount: 8 },
-  ];
+  ]);
 
   constructor() {}
 
   addIngredient(ingredient: any) {
-    this.ingredients.push(ingredient);
+    const newIngredients = [...this.ingredients.value, ingredient];
+    this.ingredients.next(newIngredients);
   }
 
   updateIngredient(ingredient: any) {
-    this.ingredients.forEach((item) => {
+    this.ingredients.value.forEach((item) => {
```

Comment:
use `.find()` to find the updating item. ex: `this.ingredients.find(ig => ig.id === ingredient.id)`
and then change its properties.